dev/run-tests-jenkins.py

```diff
@@ -0,0 +1,253 @@
+
+        rm_r(log_archive)
+
+
+def run_pr_tests(pr_tests, ghprb_actual_commit, sha1):
+    # Ensure we save off the current HEAD to revert to
+    current_pr_head = run_cmd(['git', 'rev-parse', 'HEAD'], return_output=True).strip()
+    pr_results = list()
+
+    for pr_test in pr_tests:
+        pr_results.append(run_cmd(['bash', os.path.join(SPARK_HOME, 'dev', 'tests', pr_test),
+                                   ghprb_actual_commit, sha1],
+                                  return_output=True).strip())
+        # Ensure, after each test, that we're back on the current PR
+        run_cmd(['git', 'checkout', '-f', current_pr_head])
+    return pr_results
+
+
+def bind_message_base(build_display_name, build_url, ghprb_pull_id, short_commit_hash, commit_url):
+    """
+    Given base parameters to generate a strong Github message response, binds those
+    parameters into a closure without the specific message and returns a function
+    able to generate strong messages for a specific description.
+    """
+    return lambda mssg, post_mssg="": \
+        '**[Test build ' + build_display_name + ' ' + mssg + '](' + build_url + \
+        'console)** for PR ' + ghprb_pull_id + ' at commit [\`' + short_commit_hash + '\`](' + \
+        commit_url + ')' + str(' ' + post_mssg + '.') if post_mssg else '.'
+
+
+def success_result_note(mssg):
+    return ' * This patch ' + mssg + '.'
+
+
+def failure_result_note(mssg):
+    return ' * This patch **fails ' + mssg + '**.'
+
+
+def run_tests(tests_timeout):
+    test_proc = subprocess.Popen(['timeout',
+                                  tests_timeout,
+                                  os.path.join(SPARK_HOME, 'dev', 'run-tests')]).wait()
+    test_result = test_proc.returncode
+
+    failure_note_by_errcode = {
+        ERROR_CODES["BLOCK_GENERAL"]: failure_result_note('some tests'),
+        ERROR_CODES["BLOCK_RAT"]: failure_result_note('RAT tests'),
+        ERROR_CODES["BLOCK_SCALA_STYLE"]: failure_result_note('Scala style tests'),
+        ERROR_CODES["BLOCK_PYTHON_STYLE"]: failure_result_note('Python style tests'),
+        ERROR_CODES["BLOCK_DOCUMENTATION"]: failure_result_note('to generate documentation'),
+        ERROR_CODES["BLOCK_BUILD"]: failure_result_note('to build'),
+        ERROR_CODES["BLOCK_MIMA"]: failure_result_note('MiMa tests'),
+        ERROR_CODES["BLOCK_SPARK_UNIT_TESTS"]: failure_result_note('Spark unit tests'),
+        ERROR_CODES["BLOCK_PYSPARK_UNIT_TESTS"]: failure_result_note('PySpark unit tests'),
+        ERROR_CODES["BLOCK_SPARKR_UNIT_TESTS"]: failure_result_note('SparkR unit tests'),
+        ERROR_CODES["BLOCK_TIMEOUT"]: failure_result_note('from timeout after a configured wait' +
+                                                          ' of \`' + tests_timeout + '\`')
+    }
+
+    if test_result == 0:
+        test_result_note = success_result_note('passes all tests')
+    else:
+        test_result_note = failure_note_by_errcode(test_result)
```

Comment:
`failure_note_by_errcode` is a dictionary, but here you're calling it like a function; this will result in an error.